runtime/bcutil/ROMClassCreationContext.hpp

```diff
@@ -293,7 +290,38 @@ class ROMClassCreationContext
 		}
 	}
 
-	void setIsLambda(bool isLambda) { _isLambda = isLambda; }
+	bool isUnsafeClassSharingEnabled() const {
+		/*
+		 * The command line option -XX:[+/-]ShareUnsafeClasses, combined with -XX:[+/-]ShareAnonymousClasses will have 4 different behaviours:
+		 * 1. -XX:+ShareAnonymousClasses -XX:+ShareUnsafeClasses, this will share all unsafe classes
+		 * 2. -XX:+ShareAnonymousClasses -XX:-ShareUnsafeClasses, this will only share anonymous classes and not other unsafe classes
+		 * 3. -XX:-ShareAnonymousClasses -XX:+ShareUnsafeClasses, this will only share unsafe classes that are not anonymous
+		 * 4. -XX:-ShareAnonymousClasses -XX:-ShareUnsafeClasses, this will share neither.
+		 * The current default behavior is the 1st option.
+		 */
+		bool isEnabled = false;
+
+		bool isAnonDefineClassSharingEnabled = false;
+		bool isUnsafeDefineClassSharingEnabled = false;
+		if ((NULL != _javaVM)
+		&& (NULL != _javaVM->sharedClassConfig)
```

Comment:
I don't think we need to do NULL check on _javaVM and _javaVM->sharedClassConfig here. They cannot be NULL here.